client/src/pages/Profile/index.js

```diff
@@ -1,6 +1,200 @@
+/* eslint-disable react/no-unused-state */
 import React from 'react';
+import { LockOutlined, MailOutlined, EditOutlined } from '@ant-design/icons';
+import { Button, Form, Input, notification, Spin } from 'antd';
+import Navbar from '../../components/Navbar';
+
 import './style.css';
 
-const Profile = () => <div className="Profile" />;
+class Profile extends React.Component {
+  state = {
+    loading: true,
+    profileData: {},
+    isUpdate: false,
+  };
+
+  formRef = React.createRef();
+
+  componentDidMount() {
+    this.fetchProfileData().then(() => this.setState({ loading: false }));
+  }
+
+  // nameOnChange = (e) => {
+  //   this.setState({ profileData: { name: e.target.value } });
+  // };
+
+  toggleUpdate = () => {
+    const { isUpdate } = this.state;
+    this.setState({ isUpdate: !isUpdate });
+  };
+
+  fetchProfileData = () =>
+    fetch(`/api/v1/profile`)
+      .then((res) => {
+        if (!res.ok) {
+          res.json().then(notification.error);
+          throw res.statusText;
+        }
+        return res.json();
+      })
+      .then((results) => {
+        this.formRef.current.setFieldsValue(results);
```

Comment:
You can do the same using form initial values from ant design but because you won't have 
data in the state yet what you need to do is to have a condition that checks for the `profileData` and when the data arrive then you display the from with its initial values 

```jsx
if (!state.profileData) {
  return <LoadingComponent />
}
```

I'm not saying that there is something wrong with your code but because react docs say that you should avoid using refs if you can.